### PR TITLE
BL-2800 Handle upload attachment too large

### PR DESCRIPTION
**Pull request by JohnThomson**
[Review on Reviewable](https://reviewable.io/reviews/bloombooks/bloomdesktop/831)

### CODE REVIEW

File: src/BloomExe/MiscUI/ProblemReporterDialog.cs [hunk 1]
```diff
@@ -332,6 +332,28 @@ private bool SubmitToYouTrack()
 							zip.Save();
 							AddAttachment(bookZip.Path);
 						}
+						catch (InvalidRequestException e)
+						{
+							// We get this rather unspecific exception (with the even more unhelpful message 'NoContent') if the attachment is too large.
+							// There might of course be other reasons.
+							bookZip.Detach();
```
Comment (hatton) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: Why is the Detach needed here?

Comment (JohnThomson) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: I don't know. It was present in the generic case below that handles all exceptions, so I presumed there was a good reason for it.
It's very possible that they were intended for debugging and both should be removed.

Comment (hatton) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: Yes, I an I know why the case *I* was handling needed it. But *your* case doesn't :-)

**Comment by JohnThomson** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[src/BloomExe/MiscUI/ProblemReporterDialog.cs, line 339 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/831#-K1uHFMhOzaQDEWmFPV0-r1-339)** ([raw file](https://github.com/bloombooks/bloomdesktop/blob/9506a90edbbe8c8952935fdaa2f26c64248b6e2b/src/BloomExe/MiscUI/ProblemReporterDialog.cs#L339)):</sup>
I re-arranged it a little.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/831)

**Comment by hatton** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/831)

**Pull request merged**